### ISSUE TITLE
Customize `no-await-in-loop` to allow `while`

### ISSUE THREAD

**Issue by medikoo** [eslint/eslint#10198]
**What rule do you want to change?**

[no-await-in-loop](https://eslint.org/docs/4.0.0/rules/no-await-in-loop)

**Does this change cause the rule to produce more or fewer warnings?**

Optionally fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option, that will allow to seclude `while` loops from checking.

**Please provide some example code that this change will affect:**

e.g. there are cases where we poll remote API, until some desired change occurs, with async functions it can neatly be constructed as follows:

```js
while (obj.state !== 'EXPECTED') {
  await sleep(1000);
  obj = await requestObj();
}
```

**What does the rule currently do for this code?**

Currently it unconditionally warns about using `await` in `while`.

**What will the rule do after it's changed?**

It will not warn about `await` in `while`, if rule is configured with specific option

**Comment by kaicataldo** [comment]
Sorry we lost track of this. @eslint/eslint-team Mind taking a look?
